query_frontend/src/provider.rs

```diff
@@ -72,6 +84,9 @@ pub trait MetaProvider {
 
     /// Get udaf by name.
     fn aggregate_udf(&self, name: &str) -> Result<Option<AggregateUdf>>;
+
+    /// Return all tables.
+    fn all_tables(&self) -> Result<Vec<TableRef>>;
```

Comment:
Add comments when this is used now, and warn users this maybe incur expensive cost.